### PR TITLE
fix(claims): remove fields from claims

### PR DESCRIPTION
**Pull request by hejkerooo**

### CODE REVIEW

File: src/iam.ts [hunk 1]
```diff
@@ -17,12 +17,12 @@
...
 import {
-    IRoleDefinition,
+    DomainReader,
+    EncodedCall,
     IAppDefinition,
     IOrganizationDefinition,
+    IRoleDefinition,
     PreconditionType,
-    EncodedCall,
-    DomainReader,
```
Comment (JGiter) [inline src/iam.ts:25]: Why this change?

Comment (hejkerooo) [reply src/iam.ts:25]: Sorted imports, eslint rule

File: test/staking.ts [hunk 2]
```diff
@@ -193,7 +193,7 @@ export const stakingTests = (): void => {
 
             const registrationTypes = [RegistrationTypes.OnChain];
             await patronIam.createClaimRequest({
-                claim: { claimType: `${patronRole}.${root}`, claimTypeVersion: 1, fields: [] },
```
Comment (jrhender) [inline test/staking.ts:196]: I don't understand why this improves the test? Is it because it is unrealistic to have an empty array for the `fields`?

File: test/claimsTests/enrollmentClaimsTests.ts [hunk 3]
```diff
@@ -88,7 +88,7 @@ export function enrollmentClaimsTests() {
     ) {
         const requesterDID = requesterIam.getDid();
         await requesterIam.createClaimRequest({
-            claim: { claimType, claimTypeVersion: version, fields: [] },
```
Comment (jrhender) [inline test/claimsTests/enrollmentClaimsTests.ts:91]: Same question as below, but I'm not sure how this improves the tests. Does this change the tests so that they would have failed with the previous implementation?
I think what we should actually be doing is adding data to the fields and making sure that it is stripped out when we issue

File: src/iam.ts [hunk 4]
```diff
@@ -1498,7 +1503,7 @@ export class IAM extends IAMBase {
         subject,
         registrationTypes = [RegistrationTypes.OffChain],
     }: {
-        claim: { claimType: string; claimTypeVersion: number; fields: { key: string; value: string | number }[] };
+        claim: { claimType: string; claimTypeVersion: number };
```
Comment (jrhender) [inline src/iam.ts:1506]: I think in `createClaimRequest` we want to include the `fields` so that the issuer can read them. It is only at the time of issuing that we want to remove the `fields`

File: test/iam.test.ts [hunk 5]
```diff
@@ -1,19 +1,19 @@
 import { BigNumber, utils, Wallet } from "ethers";
```
Comment (hejkerooo) [inline test/iam.test.ts:1]: No changes were made for this file - just eslint/prettier

File: test/staking.ts [hunk 6]
```diff
@@ -3,7 +3,7 @@ import { StakingPoolFactory__factory } from "../ethers/factories/StakingPoolFact
 import { RewardPool__factory } from "../ethers/factories/RewardPool__factory";
```
Comment (hejkerooo) [inline test/staking.ts:3]: No changes were made for this file - just eslint/prettier
